morango/certificates.py

```diff
@@ -137,12 +150,53 @@ def check_certificate(self):
             if not self.parent.verify(self.serialized, self.signature):
                 raise CertificateSignatureInvalid()
             # check that certificate's scope is a subset of parent's scope
-            self.get_scope().verify_subset_of(self.parent.get_scope())
+            if not self.get_scope().is_subset_of(self.parent.get_scope()):
+                raise CertificateScopeNotSubset()
             # check that certificate is for same profile as parent
             if self.profile != self.parent.profile:
                 raise CertificateProfileInvalid("Certificate profile is {} but parent's is {}" \
                                                 .format(self.profile, self.parent.profile))
 
+    @classmethod
+    def save_certificate_chain(cls, cert_chain, expected_last_id=None):
+
+        # parse the chain from json if needed
+        if isinstance(cert_chain, six.string_types):
+            cert_chain = json.loads(cert_chain)
+
+        # start from the bottom of the chain
+        cert_data = cert_chain[-1]
+
+        # create an in-memory instance of the cert from the serialized data and signature
+        cert = cls.deserialize(cert_data["serialized"], cert_data["signature"])
+
+        # verify the id of the cert matches the id of the outer serialized data
+        assert cert_data["id"] == cert.id
+
+        # check that the expected ID matches, if specified
+        if expected_last_id:
+            assert cert.id == expected_last_id
+
+        # if cert already exists locally, it's already been verified, so no need to continue
```

Comment:
Yep, otherwise the `parent` foreignkeys would have failed. We (can) only save certs for which we have the parent and it itself is already saved.